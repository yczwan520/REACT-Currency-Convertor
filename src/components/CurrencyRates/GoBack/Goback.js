import React from "react";
import { useNavigate } from "react-router-dom";

const Goback = (props) => {
  const navigate = useNavigate();
  return (
    <div>
      <p>
        This is the best currency convertor in the world! Convert foreign
        dollars to Canadian dollar
      </p>
      <p>Let's try it NOW!</p>
      <button size="huge" onClick={() => navigate(-1)}>
        Go Back
      </button>
    </div>
  );
};
export default Goback;
