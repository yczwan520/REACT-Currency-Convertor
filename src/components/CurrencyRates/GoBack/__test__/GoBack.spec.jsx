import { render, screen, waitFor } from "@testing-library/react";
import React from "react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router-dom";
import GoBack from "../Goback";
import "@testing-library/jest-dom";

describe("GoBack page", () => {
  const pageInfo =
    "This is the best currency convertor in the world! Convert foreign dollars to Canadian dollar";
  it("renders the GoBack page", () => {
    const { container } = render(
      <MemoryRouter>
        <GoBack />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText(pageInfo)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Go Back" })).toBeInTheDocument();
  });
});
