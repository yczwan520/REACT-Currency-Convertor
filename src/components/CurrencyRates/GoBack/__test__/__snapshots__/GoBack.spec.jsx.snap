// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBack page renders the GoBack page 1`] = `
<div>
  <div>
    <p>
      This is the best currency convertor in the world! Convert foreign dollars to Canadian dollar
    </p>
    <p>
      Let's try it NOW!
    </p>
    <button>
      Go Back
    </button>
  </div>
</div>
`;
